kedro/io/core.py

```diff
@@ -189,6 +192,22 @@ def from_config(
             )
         return data_set
 
+    def __init__(
+        self,
+        load_args: Optional[Dict[str, Any]] = None,
+        save_args: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        self._load_args = (
+            {**self.DEFAULT_LOAD_ARGS, **load_args}
+            if load_args is not None
+            else self.DEFAULT_LOAD_ARGS
```

Comment:
I appreciate that it's copied from some of the existing datasets but this should probably copy to prevent accidental mutation of the defaults, perhaps
```
self._load_args = dict(self.DEFAULT_LOAD_ARGS)
if load_args:
    self._load_args.update(**load_args)
```
which also has the nice effect of removing inline conditional, inline conditionals split across lines should generally be avoided

Reply:
@tolomea Thanks for the catch! This actually fixes the failing `e2e-tests` and another "weird" case where `super().__init__` was causing failing behavior.